daemon/cmd/endpoint.go

```diff
@@ -569,6 +569,12 @@ func putEndpointIDHandler(d *Daemon, params PutEndpointIDParams) (resp middlewar
 	}
 	defer r.Done()
 
+	// Wait for the host datapath to be initialized before starting to manage
+	// endpoint programs. bpf_lxc makes the assumption that bpf_host is loaded
+	// and its policy program(s) inserted. Can't live in createEndpoint as that
+	// function is called from tests that don't set up a host datapath.
+	<-d.Datapath().Loader().HostDatapathInitialized()
```

Comment:
Makes sense to me,  blocking inside the api handler does seem a bit strange to me however.

One possible improvement on this could be to `select {}` on both this and the request context in case the request is cancelled prior to this completing.

```go
select {
    case <-d.Datapath().Loader().HostDatapathInitialized():
    case <-d.apiLimiterSet.Wait(params.HTTPRequest.Context().Done():
        // ...
}
```

Reply:
The code in question has been removed.